### PR TITLE
optimize resourceconsist controller events

### PR DESCRIPTION
**Pull request by shaofan-hs**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/frame/controller/resourceconsist_controller.go [hunk 1]
```diff
@@ -152,100 +155,85 @@ func (r *Consist) Reconcile(ctx context.Context, request reconcile.Request) (rec
...
 		return reconcile.Result{}, err
 	}
 	currentEmployees, err := r.adapter.GetCurrentEmployee(ctx, employer)
 	if err != nil {
-		r.logger.Error(err, fmt.Sprintf("get current employees for %s: %s/%s failed",
-			employer.GetObjectKind().GroupVersionKind().Kind, employer.GetNamespace(), employer.GetName()))
-		r.recorder.Eventf(employer, corev1.EventTypeWarning, "GetCurrentEmployeeFailed",
-			"get current employees failed: %s", err.Error())
+		logger.Error(err, "get current employees failed")
 		return reconcile.Result{}, err
```
Comment (WeichengWang1) [inline pkg/frame/controller/resourceconsist_controller.go:213]: should we ignore this event?

File: pkg/frame/controller/resourceconsist_controller.go [hunk 2]
```diff
@@ -256,14 +244,10 @@ func (r *Consist) Reconcile(ctx context.Context, request reconcile.Request) (rec
...
-			r.logger.Error(err, fmt.Sprintf("record status for %s: %s/%s failed",
-				employer.GetObjectKind().GroupVersionKind().Kind, employer.GetNamespace(), employer.GetName()))
-			r.recorder.Eventf(employer, corev1.EventTypeWarning, "recordStatusesFailed",
-				"record status for employer failed: %s", err.Error())
+			logger.Error(err, "record status failed")
 			return reconcile.Result{}, err
 		}
 	}
 
-	r.recorder.Eventf(employer, corev1.EventTypeNormal, "ReconcileSucceed", "")
```
Comment (WeichengWang1) [inline pkg/frame/controller/resourceconsist_controller.go:267]: I noticed that some events ignored. Could u explain why those events ignored and how to choose what events should be remained?

Comment (shaofan-hs) [reply pkg/frame/controller/resourceconsist_controller.go:267]: Generally speaking, event is for common users, and log is for controller developers, we can separate them for better performance. If we meet a problem, we can firstly view the events, then if we need more detail message,  we can search the logs.

**Review by WeichengWang1 — APPROVED** [review]
lgtm

**Pull request merged**